pkg/apis/core/v1alpha1/types.go

```diff
@@ -250,6 +250,45 @@ const (
 // DataVolumeCloneSourceSubresource is the subresource checked for permission to clone
 const DataVolumeCloneSourceSubresource = "source"
 
+//CDIStorageClassCapabilities provides a CDI specific extension for StorageClasses
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName=cdistorageclass;cdistorageclasses,scope=Cluster
+type CDIStorageClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CDIStorageClassSpec   `json:"spec"`
+	Status CDIStorageClassStatus `json:"status,omitempty"`
+}
+
+//CDIStorageClassStatusSpec defines specification for StorageClasses Capabilities
+type CDIStorageClassSpec struct {
+	// PvcProfiles
+	PvcProfiles []string `json:"pvcProfiles"`
```

Comment:
I would call this VolumeProfile (no s) a type that contains some fields something like, (copied these from the PVC struct):
```go
type CDIStorageClassSpec struct {
  // volumeProfiles
  volumeProfiles []VolumeProfile `json:"volumeProfiles,omitempty"`
}

type VolumeProfile struct {
  // AccessModes contains the desired access modes the volume should have.
  // More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
  // +optional
  AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes,omitempty" protobuf:"bytes,1,rep,name=accessModes,casttype=PersistentVolumeAccessMode"`
  // volumeMode defines what type of volume is required by the claim.
  // Value of Filesystem is implied when not included in claim spec.
  // +optional
  VolumeMode *corev1.PersistentVolumeMode `json:"volumeMode,omitempty" protobuf:"bytes,6,opt,name=volumeMode,casttype=PersistentVolumeMode"`
}
```
Note the omitempty in the "json:" section. That means the value is optional and can be missing.

Reply:
thanks for the sample,
and renamed ClaimPropertySet